### ISSUE TITLE
eslint stopped ignoring node_modules

### ISSUE THREAD

**Issue by majelbstoat** [Microsoft/vscode-eslint#503]
I have the following in `.eslintignore` at the root of the workspace:

```
**/node_modules/**
**/generated/schema.ts
```

Until recently this correctly ignored all expected files in the workspace. For example, it would correctly ignore all of:

```
/service/a/generated/schema.ts
/service/a/node_modules/apollo/graphql/apolloserver.ts
/service/b/node_modules/second/module/more.ts
```

As of vscode 1.25, (maybe slightly earlier, but definitely not in vscode 1.23), the exact same definition now only ignores:

```
/service/a/generated/schema.ts
```

and problems are reported for code under `/service/a/node_modules` and `/service/b/node_modules`. 

I have confirmed that removing `**/generated/schema.ts` from `.eslintignore` causes problems to be shown in `/service/a/generated/schema.ts`, so the ignore file is being processed. However, no variation that I have tried causes code under the `node_modules` folders to be excluded.

The problems are coming from `eslint`, not `ts`, or another linter, as far as I can tell. As an example:

`[eslint] Delete `;` (prettier/prettier)`

I do not have eslint.working_directories set, and am doing the exclusions from the root. This was the same as when it was fully working.

**Comment by majelbstoat** [comment]
After further debugging, it appears that the issue is because of this bug: https://github.com/Microsoft/vscode/issues/12448 which causes "jump to definition" to open a file with the entire path defined as lowercase. For some reason, this causes the eslintignore definition not to be respected. 

When I open `service/a/node_modules/apollo/graphql/ApolloServer.ts` manually, the file is not linted, as expected. However, when I cmd+click to jump to a definition inside that file, it is being opened as `service/a/node_modules/apollo/graphql/apolloserver.ts` the file is linted.

**Comment by dbaeumer** [comment]
@majelbstoat thanks for debugging this. Actually what you are describing explains why it fails. The reason is as follows: if `service/a/node_modules/apollo/graphql/apolloserver.ts` is opened as an absolute path if doesn't belong to a workspace folder. Then the ESLint extension doesn't know which directory is the current working directory (which is the workspace folder). The eslint npm module reads the .eslintignore file relative to the current working directory. In the above case it doesn't find one hence the file is validated.

There is actually not much the ESLint extension can do about it since it can't correctly infer the working directory to use.

**Comment by sibelius** [comment]
I think this is a bug on eslint, not vscode

this is happening to me as well, but on command line

when using yarn workspaces

**Comment by dbaeumer** [comment]
@sibelius yes agree. That is why I marked the issue as upstream :-) but forgot to close it. Thanks for reminding.
